src/server/worker.cc

```diff
@@ -76,17 +78,40 @@ Worker::Worker(Server *srv, Config *config) : srv(srv), base_(event_base_new())
 }
 
 Worker::~Worker() {
-  std::vector<redis::Connection *> conns;
-  conns.reserve(conns_.size() + monitor_conns_.size());
-
-  for (const auto &iter : conns_) {
-    conns.emplace_back(iter.second);
-  }
-  for (const auto &iter : monitor_conns_) {
-    conns.emplace_back(iter.second);
+  auto conns = tbb::parallel_reduce(
+      conns_.range(), std::vector<redis::Connection *>{},
+      [](const ConnMap::range_type &range, std::vector<redis::Connection *> &&result) {
+        for (auto &it : range) {
+          result.push_back(it.second);
+        }
+        return result;
+      },
+      [](const std::vector<redis::Connection *> &lhs, const std::vector<redis::Connection *> &rhs) {
+        std::vector<redis::Connection *> result = lhs;
+        result.insert(result.end(), rhs.begin(), rhs.end());
+        return result;
+      });
+
+  auto monitor_conns = tbb::parallel_reduce(
+      monitor_conns_.range(), std::vector<redis::Connection *>{},
+      [](const ConnMap::range_type &range, std::vector<redis::Connection *> &&result) {
+        for (auto &it : range) {
+          result.push_back(it.second);
+        }
+        return result;
+      },
+      [](const std::vector<redis::Connection *> &lhs, const std::vector<redis::Connection *> &rhs) {
+        std::vector<redis::Connection *> result = lhs;
+        result.insert(result.end(), rhs.begin(), rhs.end());
+        return result;
+      });
```

Comment:
```suggestion
  auto collect_conns_fn = [](ConnMap &conns) {
    return parallel_reduce(
        conns.range(), std::vector<redis::Connection *>{},
        [](const ConnMap::range_type &range, std::vector<redis::Connection *> &&result) {
          for (auto &it : range) {
            result.push_back(it.second);
          }
          return result;
        },
        [](const std::vector<redis::Connection *> &lhs, const std::vector<redis::Connection *> &rhs) {
          std::vector<redis::Connection *> result = lhs;
          result.insert(result.end(), rhs.begin(), rhs.end());
          return result;
        });
  };

  auto conns = collect_conns_fn(conns_);
  auto monitor_conns = collect_conns_fn(monitor_conns_);
```

Reply:
Hi @git-hulk ,

Thanks very much for your suggestions! 😊
I have refactored the duplicated code.

Best Regards,
Edward